codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -492,45 +481,105 @@ impl EventTemplate {
                 let topic_count = params
                     .iter()
                     .fold(1, |acc, param| if param.indexed { acc + 1 } else { acc });
+                let sighash = config_event.sighash.to_string();
+                let event_name = config_event.name.capitalize();
+                let data_type = data_type_expr.to_string();
+                let params_raw_event_schema =
+                    data_type_expr.to_rescript_schema(&"eventArgs".to_string());
+                let convert_hyper_sync_event_args_code =
+                    Self::generate_convert_hyper_sync_event_args_code(params);
+                let event_filter_type = Self::generate_event_filter_type(params);
+                let get_topic_selection_code = Self::generate_get_topic_selection_code(params);
+
+                let module_code = format!(
+                    r#"
+let sighash = "{sighash}"
+let topicCount = {topic_count}
+let name = "{event_name}"
+let contractName = contractName
+
+@genType
+type eventArgs = {data_type}
+let paramsRawEventSchema = {params_raw_event_schema}
+let convertHyperSyncEventArgs = {convert_hyper_sync_event_args_code}
+
+let handlerRegister: HandlerTypes.Register.t<eventArgs> = HandlerTypes.Register.make(
+  ~topic0=sighash->EvmTypes.Hex.fromStringUnsafe,
+  ~contractName,
+  ~eventName=name,
+)
+
+@genType
+type eventFilter = {event_filter_type}
+
+let getTopicSelection = {get_topic_selection_code}"#
```

Comment:
Can we deduplicate this format template? Maybe create a function above the match statement that produces this format with the given params?